test/e2e/framework/exec_utils.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/url"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	"github.com/onsi/gomega"
+)
+
+// ExecOptions passed to ExecWithOptions
+type ExecOptions struct {
+	Command       []string
+	Namespace     string
+	PodName       string
+	ContainerName string
+	Stdin         io.Reader
+	CaptureStdout bool
+	CaptureStderr bool
+	// If false, whitespace in std{err,out} will be removed.
+	PreserveWhitespace bool
+	Quiet              bool
+}
+
+// ExecWithOptions executes a command in the specified container,
+// returning stdout, stderr and error. `options` allowed for
+// additional parameters to be passed.
+func ExecWithOptions(f *Framework, options ExecOptions) (string, string, error) {
```

Comment:
Can we use `util.ExecuteCommandInContainer()` directly?

Reply:
When I was referencing `kubernetes/test/e2e/network/networking_perf.go` to write e2e, I also considered this issue. If it is in the e2e scenario, I think it would be better to use `func ExecCommandInContainer(f *framework.Framework, podName, containerName string, cmd ...string) string`, because I can use `framework.Logf` to output some debugging information and obtain some properties through `f.`.

Reply:
`klog.Infof()` can also print debug information.

What does `obtain some properties through f.` mean?

Reply:
Maybe I can pass `f.Namespace.Name` as a parameter.
Thank you for your suggestion. I will try to modify `exec_util.go` and reuse the code from `/pkg/util/pod_exec.go`.